app/src/main/java/com/nmakademija/nmaakademija/api/FirebaseRealtimeApi.java

```diff
@@ -0,0 +1,144 @@
+                        }
+                    }
+
+                    apiLoadedListener.onAcademicsLoaded(filteredAcademics);
+                }
+            }
+
+            @Override
+            public void onAcademicsLoadingFailed(Exception exception) {
+                AcademicsLoadedListener apiLoadedListener = loadedListener.get();
+                if (apiLoadedListener != null) {
+                    apiLoadedListener.onAcademicsLoadingFailed(exception);
+                }
+            }
+        });
+    }
+
+    public static void getSections(SectionsLoadedListener listener) {
+        final WeakReference<SectionsLoadedListener> loadedListener =
+                new WeakReference<>(listener);
+
+        FirebaseDatabase.getInstance().getReference("sections")
+                .addListenerForSingleValueEvent(
+                        new ApiReferenceListener<Section>(Section.class) {
+
+                            @Override
+                            public void onLoaded(ArrayList<Section> sections) {
+                                SectionsLoadedListener apiLoadedListener = loadedListener.get();
+                                if (apiLoadedListener != null) {
+                                    apiLoadedListener.onSectionsLoaded(sections);
+                                }
+                            }
+
+                            @Override
+                            public void onFailed(Exception ex) {
+                                SectionsLoadedListener apiLoadedListener = loadedListener.get();
+                                if (apiLoadedListener != null) {
+                                    apiLoadedListener.onSectionsLoadingFailed(ex);
+                                }
+                            }
+
+                        });
+    }
+
+    public static void getSchedules(SchedulesLoadedListener listener, final int sectionId) {
+        final WeakReference<SchedulesLoadedListener> loadedListener =
+                new WeakReference<>(listener);
+
+        FirebaseDatabase.getInstance().getReference("schedules")
+                .addListenerForSingleValueEvent(
+                        new ApiReferenceListener<ScheduleEvent>(ScheduleEvent.class) {
+
+                            @Override
+                            public void onLoaded(ArrayList<ScheduleEvent> scheduleEvents) {
+                                SchedulesLoadedListener apiLoadedListener = loadedListener.get();
+                                if (apiLoadedListener != null) {
+                                    ArrayList<ScheduleEvent> filteredSchedules = new ArrayList<>();
+
+                                    for (ScheduleEvent scheduleEvent : scheduleEvents) {
+                                        if (scheduleEvent.getSectionId() == 0 ||
+                                                scheduleEvent.getSectionId() == sectionId) {
+                                            filteredSchedules.add(scheduleEvent);
```

Comment:
This check doesn't work correctly. We need to check it. `scheduleEvent::getSectionId` always returns 0.